rewriter/Minitest.cc

```diff
@@ -368,13 +467,27 @@ ast::ExpressionPtr prepareTestEachBody(core::MutableContext ctx, core::NameRef e
 }
 
 ast::ExpressionPtr runSingle(core::MutableContext ctx, bool isClass, ast::Send *send, bool insideDescribe) {
+    // Handle include_context first (1+ pos args + optional kwargs/block)
+    if (send->fun == core::Names::includeContext() && send->numPosArgs() >= 1) {
+        auto &arg = send->getPosArg(0);
+        auto argString = to_s(ctx, arg);
+        // Create an include statement that includes the shared_examples companion module
+        // Use proper scoping - if we're inside a describe block, look locally; otherwise use root
+        auto moduleName = makeSharedExamplesModuleConstantLocal(ctx, arg.loc(), argString);
+        // Create an include statement to include the shared_examples companion module
+        return ast::MK::Send1(send->loc, ast::MK::Self(send->loc), core::Names::include(), send->funLoc,
+                              std::move(moduleName));
+    }
+
     if (!send->hasBlock()) {
         return nullptr;
     }
 
     auto *block = send->block();
 
-    if (!send->recv.isSelfReference()) {
+    if (!send->recv.isSelfReference() && !((isDescribeOrSimilar(send->fun) || isSharedExamples(send->fun) ||
+                                            send->fun == core::Names::includeContext()) &&
```

Comment:
Is it intentional to have `includeContext` here?

It looks like you have "Handle include_context first" above, and there are no test for `RSpec.include_context`, so I think that handling includeContext here is dead.

Reply:
removed in c198d7a84 thank you